### PR TITLE
Sinclair pull request

### PR DESCRIPTION
**Pull request by SMSinclair**

### CODE REVIEW

**Review by hughjafro — COMMENTED** [review]
Module 1:

- Start a clean notebook.
  - I like your text blocks that help to explain process
- Do train/validate/test split with the Tanzania Waterpumps data.
  - Stretch: Explore different train_size and test_size
- Begin to explore and clean the data.
  - Import has the correct shape
  - Good methodical approach to analyze each feature
  - Good start to analyze for cardinality
- Select different numeric and categorical features.
  - The Permit feature has a quite a few missing values
  - Your cat features selections are a good choice
- Do one-hot encoding. (Remember it may not work with high cardinality categoricals.)
- Scale features.
- Use scikit-learn for logistic regression.
  - Check
  - Does using `solver='liblinear' in your LR model make a difference in score?
- Get your validation accuracy score.
  - nice score
- Get and plot your coefficients.
  - How you address and impute missing/zero values have an effect on model scores, try different ways
- Submit your predictions to our Kaggle competition.
- Commit your notebook to your fork of the GitHub repo.

**Review by hughjafro — COMMENTED** [review]
**Module2 Decision Tree:**
- Good exploration of different models
- Sometimes when your model is running for 3 hours, you might need to take a walk
- Interesting to explore why your GB score is lower than RF, because as long as your parameters are carefully tuned GB should perform better.
  - GB is more prone to overfitting (higher variance) and generally takes longer because the trees are built one at a time/sequentially. Takes a bit of work to tune properly, but once you do the results are generally better than RF.
  - RF is good for multiclass problems and is easier to tune than GB. When there are categoricals, RF will tend to bias those cats with more levels (partial permutation methods help with this)

The following is a good read: [RF or GB?](https://www.quora.com/When-would-one-use-Random-Forests-over-Gradient-Boosted-Machines-GBMs)

**Review by hughjafro — COMMENTED** [review]
### Module 03 Random Forest:

**Overall:** Awesome Job! Soooo close to the top.
**Feedback:** Try more feature engineering and how you and how you address NaN's.

_Do more exploratory data analysis, data cleaning, feature engineering, and feature selection._
- Would separating certain categories into 2/3 parts affect your score?

_Try a Random Forest Classifier._
- Check

_Try Ordinal Encoding._
- Check

**Submit new predictions.**

**Comment by hughjafro** [comment]
### Module 04 Confusion Matrix:

**Overall:** Great job this week
**Feedback:** Your code is always well organized and easy to read.

_Get and interpret the confusion matrix for classification models_
- At least take a stab at the confusion matrix
- What does the c-matrix tell you?
- What is the significance of the precision, recall, f-1 scores?

_Kaggle Challenge_
- Why do you think your score dropped between public and private board?
- What can be done to mitigate overfitting?
- Ask yourself, what you would do differently on your next Kaggle challenge? In approach and execution.
- Don't give up on xgboost 🚀. This is a kaggle mainstay.

**Great Job**
